pyodide_build/pywasmcross.py

```diff
@@ -8,14 +8,21 @@
 cross-compiling and then pass the command long to emscripten.
 """
 
+import functools
 import json
 import os
+import re
+import shutil
+import subprocess
 import sys
+import tempfile
+from collections.abc import Iterable, Iterator
 from pathlib import Path
+from typing import Literal, NamedTuple
 
 from __main__ import __file__ as INVOKED_PATH_STR
 
-INVOKED_PATH = Path(INVOKED_PATH_STR)
+SHAREDLIB_REGEX = re.compile(r"\.so(.\d+)*$")
```

Comment:
I haven't profiled how expensive this regular expression is or how much it is called – perhaps a thousand or so times in total as it's called at link time? It would make sense to store it as a variable.